NumericOutcome.md

```diff
@@ -0,0 +1,542 @@
+Additionally it leads to extra complexity in contract construction.
+This design's only benefit is that it results in simpler (although larger) fraud proofs.
+
+The large multi-signature design was abandoned because the above proposal is sufficient to generate fraud proofs.
+If an oracle incorrectly signs for an event, then only the sum of the digit signatures `s(1..m)`
+is recoverable on-chain using the adaptor signature which was given to one's counter-party.
+This sum is sufficient information to determine what was signed however as one can iterate through
+all possible composite adaptor points until they find one whose pre-image is the signature sum found on-chain.
+This will determine what digits `(d1, ..., dm)` were signed and these values along with the oracle
+announcement and `s(1..m)` is sufficient information to generate a fraud proof in the multi-nonce setting. 
+
+## Contract Execution Transaction Compression
+
+Anytime there is a range of numeric outcomes `[start, end]` which result in the same payouts for all parties,
+then a compression function can be run to reduce the number of CETs to be `O(log(L))` instead of `L` where 
+`L = end - start + 1` is the length of the interval being compressed.
+
+Most contracts are expected to be concerned with some subset of the total possible domain and every
+outcome before or after that range will result in some constant maximal or minimal payout.
+This means that compression will drastically reduce the number of CETs to be of the order of the size
+of the probable domain, with further optimizations available when parties are willing to do some rounding.
+
+The compression algorithm takes as input a range `[start, end]`, a base `B`, and the number of digits
+`n` (being signed by the oracle) and returns an array of arrays of integers (which will all be in the range `[0, B-1]`).
+An array of integers corresponds to a single event equal to the concatenation of these integers (interpreted in base `B`).
+
+### Concrete Example
+
+Before generalizing or specifying the algorithm, let us run through a concrete example.
+
+We will consider the range `[135677, 138621]`, in base `10`.
+Note that they both begin with the prefix `13` which must be included in every CET, for this purpose I omit these digits for
+the remainder of this example as we can simply examine the range `[5677, 8621]` and prepend a `13` to all results.
+
+To cover all cases while looking at as few digits as possible in this range we need only consider
+`5677`, `8621` and the following cases:
+
+```
+5678, 5679,
+568_, 569_,
+57__, 58__, 59__,
+
+6_, 7_,
+
+80__, 81__, 82__, 83__, 84__, 85__,
+860_, 861_,
+8620
+```
+
+where `_` refers to an ignored digit (an omission from the array of integers).
+(Recall that all of these are prefixed by `13`).
+Thus, we are able to cover the entire interval of `2944` outcomes using only `20` CETs!
```

Comment:
I seem to have completely missed what you are talking about. You care about each of the outcomes which all lead to a different distribution of funds. How can you express 2944 different payouts in only 20 CETs!?

Reply:
Consider the range `[135677, 138621]`.
Let's take a random number in the interval `136412`.

If you encrypted the signature of a CET with a pubkey being the sum of the sigpoints of the three first digit being `136`, then ANY attestation signing for those three digits can be used to decrypt the CET.
And this is fine because `136***` is always in the wanted interval, regardless of the other `***` signatures.

Basically he disregards the less significant digits when he knows for sure that whichever they are, the attestation would still fall into the range.

Reply:
@LLFourn The only ranges being considered for compression have a constant payout value, perhaps this should be made more clear?

Reply:
From the beginning of this section:
```
Anytime there is a range of numeric outcomes `[start, end]` which result in the same payouts for all parties,
then a compression function can be run to reduce the number of CETs to be `O(log(L))` instead of `L` where 
`L = end - start + 1` is the length of the interval being compressed.
```

Reply:
Ok thanks. My bad. I didn't get the `same payout` bit! I see now where you are going with this.

Perhaps try and explain that applying the compression algorithm involves first splitting the the range of interest into buckets with the same payout. You then use your trick to compress the number of CETs in a bucket.